### PR TITLE
Added  field for remove method

### LINKED ISSUE #4907: remove: needs ability to be remove by rowIndex
```
	    key: "remove",
	    value: function remove(params) {
	      var len = this.options.data.length;
	      var i;
	      var row;

	      if (!params.hasOwnProperty('field') || !params.hasOwnProperty('values')) {
	        return;
	      }

		  **if (params.field.includes('rowIndex'))
			  params.values.forEach(rowIndex => this.options.data.splice(rowIndex, 1));**
		  else
	      for (i = len - 1; i >= 0; i--) {
	        row = this.options.data[i];

	        if (!row.hasOwnProperty(params.field)) {
	          continue;
	        }

	        if (params.values.includes(row[params.field])) {
	          this.options.data.splice(i, 1);

	          if (this.options.sidePagination === 'server') {
	            this.options.totalRows -= 1;
	          }
	        }
	      }

	      if (len === this.options.data.length) {
	        return;
	      }

	      this.initSearch();
	      this.initPagination();
	      this.initSort();
	      this.initBody(true);
	    }
```

### PR DESCRIPTION
**Pull request by wenzhixin**
Fix #4907, #564

Example: https://live.bootstrap-table.com/code/wenzhixin/1992

### CODE REVIEW

File: site/docs/api/methods.md [hunk 1]
```diff
@@ -381,7 +381,7 @@ The calling method syntax: `$('#table').bootstrapTable('method', parameter)`.
 
   Remove data from table, the params contain two properties:
 
-  * `field`: the field name of remove rows.
+  * `field`: the field name of remove rows. If `$index` dose not in your fields, you can use this special field `$index` to remove rows by row index.
```
Comment (djhvscf) [inline site/docs/api/methods.md]: Typo dose

Comment (djhvscf) [reply site/docs/api/methods.md]: Also, this should include that we can pass an array of indexes

File: site/docs/api/methods.md [hunk 2]
```diff
@@ -381,8 +381,8 @@ The calling method syntax: `$('#table').bootstrapTable('method', parameter)`.
 
   Remove data from table, the params contain two properties:
 
-  * `field`: the field name of remove rows.
-  * `values`: the array of values for rows which should be removed.
+  * `field`: the field name of remove rows. If `$index` does not in your fields, you can use this special field `$index` to remove rows by row index.
```
Comment (djhvscf) [inline site/docs/api/methods.md]: Hi @wenzhixin I think the correct statement should be something like 

If `$index` **is not** in your fields, you can use this special field `$index` to remove rows by row index.

**Comment by wenzhixin** [comment]
Updated

**Pull request merged**